src/main/java/com/couchbase/lite/replicator/ChangeTracker.java

```diff
@@ -569,4 +582,24 @@ public void setUsePOST(boolean usePOST) {
 
     }
 
+
+    public void setPaused(boolean paused) {
+        Log.v(Log.TAG, "setPaused: " + paused);
+        synchronized (pausedObj) {
+            this.paused = paused;
+            pausedObj.notifyAll();
+        }
+    }
+
+    protected void waitIfPaused(){
```

Comment:
I originally considered to call `waitIfPaused()`  method beginning of while loop. However response from REST api might contain multiple changes. So I decided to call `waitIfPaused()` in receivedChange(). `receivedChange()` method could be called from receivedPollResponse() if mode is LongPoll, and it is called from `runLoop()` method if mode is not LongPoll.
```
if LongPoll:
   receivedPollResponse()
      receivedChange()
         waitIfPaused()
else:
   receivedChange()
      waitIfPaused()
```